File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/ApplianceModel.java
/*
 * Copyright (c) 2018 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models;

import android.support.annotation.NonNull;

import java.io.Serializable;

/**
 * Model(プリセット情報)
 */
public class ApplianceModel implements Serializable {
    /** プリセットID */
    public String id;
    /** メーカー */
    public String manufacturer;
    /** リモコンの型番 */
    public String remote_name;
    /** シリーズ */
    public String series;
    /** プリセット名 */
    public String name;
    /** アイコン */
    public String image;

    @NonNull
    @Override
    public String toString() {
        return "ApplianceModel{" + "id='" + id + '\'' + ", manufacturer='" + manufacturer + '\'' + ", remote_name='" + remote_name + '\'' + ", series='" + series + '\'' + ", name='" + name + '\'' + ", image='" + image + '\'' + '}';
    }
}

File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/ApplianceModel.kt
/*
 * Copyright (c) 2019 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fishwaffle.natureremo.controller.NatureRemo.isIgnoreUnknown
import java.io.Serializable

/**
 * Model(プリセット情報)
 * @param id プリセットID
 * @param manufacturer メーカー
 * @param remote_name リモコンの型番
 * @param series シリーズ
 * @param name プリセット名
 * @param image アイコン
 */
@JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
data class ApplianceModel(var id: String?, var manufacturer: String?, var remote_name: String?, var series: String?, var name: String?, var image: String?) : Serializable